### ISSUE TITLE
publish new python version to pypi index

### ISSUE THREAD

**Issue by Jimexist** [apache/arrow-datafusion#480]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
(This section helps Arrow developers understand the context and *why* for this feature, in addition to  the *what*)

so far the version is too old and i cannot even get it working on my Mac (Python 3.9).

```
❯ pip3 install datafusion
Collecting datafusion
  Using cached datafusion-0.2.0.tar.gz (14 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/local/opt/python@3.9/bin/python3.9 /usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /var/folders/mn/3932bp3n1hs4cs2nqxj89x_h0000gn/T/tmp8dqh8tk3
         cwd: /private/var/folders/mn/3932bp3n1hs4cs2nqxj89x_h0000gn/T/pip-install-b0apfygw/datafusion_d45fef4b2512497aafc93868ead16560
    Complete output (16 lines):
    💥 maturin failed
      Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
      Caused by: `cargo metadata` exited with an error:     Updating git repository `https://github.com/jorgecarleitao/arrow.git`
    error: failed to get `arrow` as a dependency of package `datafusion v0.2.0 (/private/var/folders/mn/3932bp3n1hs4cs2nqxj89x_h0000gn/T/pip-install-b0apfygw/datafusion_d45fef4b2512497aafc93868ead16560)`

    Caused by:
      failed to load source for dependency `arrow`

    Caused by:
      Unable to update https://github.com/jorgecarleitao/arrow.git?rev=147aae4

    Caused by:
      revspec '147aae4' not found; class=Reference (4); code=NotFound (-3)
    Error running maturin: Command '['maturin', 'pep517', 'write-dist-info', '--metadata-directory', '/private/var/folders/mn/3932bp3n1hs4cs2nqxj89x_h0000gn/T/pip-modern-metadata-fe8ml8pt', '--interpreter', '/usr/local/Cellar/python@3.9/3.9.5/bin/python3.9']' returned non-zero exit status 1.
    Checking for Rust toolchain....
    Running `maturin pep517 write-dist-info --metadata-directory /private/var/folders/mn/3932bp3n1hs4cs2nqxj89x_h0000gn/T/pip-modern-metadata-fe8ml8pt --interpreter /usr/local/Cellar/python@3.9/3.9.5/bin/python3.9`
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/50/a7/093c2785922dc16911702fea2d2e4dbdb66f800c61573ab5808d309a26b2/datafusion-0.2.0.tar.gz#sha256=3c1849319336771fffc4310f0cebeb6bce0937ffc4bb3e4a937943177bd0ab78 (from https://pypi.org/simple/datafusion/). Command errored out with exit status 1: /usr/local/opt/python@3.9/bin/python3.9 /usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /var/folders/mn/3932bp3n1hs4cs2nqxj89x_h0000gn/T/tmp8dqh8tk3 Check the logs for full command output.
```

**Describe the solution you'd like**

release newer versions

**Describe alternatives you've considered**

**Additional context**

**Comment by jorgecarleitao** [comment]
https://github.com/apache/arrow-datafusion/issues/226 is a regression in the API over the last released version. We could also break backward compatibility and bump to 0.3, though.

**Comment by Jimexist** [comment]
maybe related https://github.com/apache/arrow-datafusion/pull/483

**Comment by parisni** [comment]
any news on pulishing 0.3.0 on pypi ?

**Comment by houqp** [comment]
@parisni it will be published as part of datafusion 5.1.0 release, which is what we are working on at the moment.

**Comment by houqp** [comment]
oh god, can't believe this took 2 months :D 0.4.0 is in PyPI now.
